src/useMap.ts

```diff
@@ -0,0 +1,83 @@
+import useForceUpdate from './useForceUpdate'
+import { useRef } from 'react'
+import useWillUnmount from './useWillUnmount'
+
+export class ObservableMap<K, V> extends Map<K, V> {
+  private readonly listeners = new Set<Function>()
+
+  observe(listener: (map: this) => void): () => void {
+    this.listeners.add(listener)
+    return () => {
+      this.listeners.delete(listener)
+    }
+  }
+
+  unobserve() {
+    this.listeners.clear()
+  }
+
+  private emit() {
+    // the ctor init will call set() so the listener Set isn't set up yet
+    if (!this.listeners) return
```

Comment:
not that it matters, but the `Set` will always be truth-y

Reply:
it won't! super() in the ctor fires before you can assign the Set to `this.listeners`